chain-core/src/init/config.rs

```diff
@@ -72,12 +73,32 @@ impl SlashRatio {
         }
     }
 
+    pub fn get_proportional(
+        self,
+        failing_validators: usize,
+        total_validators: usize,
+    ) -> SlashRatio {
+        let millis =
+            (((3.0 * failing_validators as f64) / total_validators as f64).powi(2) * 1000.0) as u64;
+        SlashRatio::new(Milli::from_millis(std::cmp::min(millis, 1000))).unwrap() * self
+    }
+
     #[inline]
     pub(crate) fn as_millis(self) -> u64 {
         self.0.as_millis()
     }
 }
 
+impl Mul for SlashRatio {
+    type Output = Self;
+
+    #[inline]
+    fn mul(self, rhs: SlashRatio) -> Self::Output {
+        SlashRatio::new(self.0 * rhs.0)
```

Comment:
can it be assertion?

Reply:
I think having `expect` is similar to having runtime assertion.And in this case, we cannot have a compile time assertion.